crates/rerun_c/src/lib.rs

```diff
@@ -112,19 +112,18 @@ pub struct CStoreInfo {
 pub struct CDataCell {
     pub component_name: CStringView,
 
-    /// Length of [`Self::bytes`].
-    pub num_bytes: u64,
+    pub array: arrow2::ffi::ArrowArray,
 
-    /// Data in the Arrow IPC encapsulated message format.
-    pub bytes: *const u8,
+    /// TODO(andreas): Use a schema registry.
+    pub schema: arrow2::ffi::ArrowSchema,
```

Comment:
Same comment here; I'd expect this to be a static pointer?

Reply:
I was actually planning to have the registry side over on Rust and give out a handle: Arrow2 doesn't take ownership of the `ArrowSchema` and instead builds up a new Rust data structure. So I figure if we cache schemas then let's go all the way.
I'll do so in a follow-up PR